lxd/storage/drivers/driver_zfs.go

```diff
@@ -685,19 +685,21 @@ func (d *zfs) patchDropBlockVolumeFilesystemExtension() error {
 	}
 
 	for _, volume := range strings.Split(out, "\n") {
-		fields := strings.SplitN(volume, "/", 3)
+		fields := strings.Split(volume, "/")
 
-		if len(fields) != 3 {
+		if len(fields) < 3 {
```

Comment:
Doesnt this just now skip volumes that are in a pool which isn't at the root of the zfs hierarchy?
So those image volumes wont get fixed?

Instead would it better to split on the ` fmt.Sprintf("%s/images", poolName)` point of the volume name, and then perform the split on `/` on the right hand part of that?

Reply:
I don't think the code is wrong, but I've changed it to `strings.SplitN(volume, fmt.Sprintf("%s/images/", poolName), 2)` just to be safe.